enhancements/api-review/add-unqualified-search-registries.md

```diff
@@ -0,0 +1,193 @@
+
+### Risks and Mitigations
+
+No new risks are introduced. The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file on all nodes in a cluster. This enhancement just adds the ability for the user to be able to configure this list. A small api change for adding the `UnqualifiedSearchRegistry` field to the images.config.openshift.io CR is needed.
+
+## Design Details
+
+### Test Plan
+
+Update the tests that are currently in the MCO to verify that `unqualified-search-registries` have been updated when the cluster wide Image CR is edited to configure this.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and Downgrades should not be affected. The `unqualified-search-registries` field already exists in the `/etc/containers/registries.conf` file. We are just exposing this option to the user so that they can configure it through the cluster wide Image CR.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+## Alternatives
+
+
```

Comment:
i think we need to explain in more detail when we would recommend usage of this feature.

there are drawbacks to supporting unqualified search registries which is why it was not included in 4.x.

the primary customer use case i have heard is desiring a single location to edit to pull images from an image registry that is closest to the cluster location without having to have some geographic aware dns router.

Reply:
The primary case seems to be when customers want to use multiple internal registries to pull images from, but the internal registries DNS could change. So pulling by short name is easier than having to go and change the image spec every time this happens.
I updated the motivation and user story with this scenario as well.
We could document the drawbacks of doing this and warn users to be cautious when using this feature.

Reply:
This is the one that stuck out in my mind
https://github.com/openshift/origin/issues/18932
https://bugzilla.redhat.com/show_bug.cgi?id=1561989

Had to do with docker though. I was a nightmare.  We tried to fix it once and regressed the world.  That was fun.

Because cri-o itself would do the search _and_ has access all the credentials for each registry, I think this would be ok in OCP 4.

Reply:
Noted in the enhancement.